controllers/dscinitialization/utils.go

```diff
@@ -440,3 +444,41 @@ func (r *DSCInitializationReconciler) createUserGroup(ctx context.Context, dscIn
 
 	return nil
 }
+
+func IsDefaultAudiences(specAudiences *[]string) bool {
+	return specAudiences == nil || reflect.DeepEqual(*specAudiences, defaultAudiences)
+}
+
+// GetEffectiveClusterAudiences returns the audiences defined for the cluster, falling back to the default audiences if needed.
+func GetEffectiveClusterAudiences(specAudiences *[]string, cli client.Client, log logr.Logger) []string {
+	if IsDefaultAudiences(specAudiences) {
+		return fetchClusterAudiences(cli, log)
+	}
+	return *specAudiences
+}
+
+func fetchClusterAudiences(cli client.Client, log logr.Logger) []string {
```

Comment:
i leave the final review to @bartoszmajsak , tbh i am not familiar with this "token" part